### ISSUE TITLE
Data formats in EDAM ontology

### ISSUE THREAD

**Issue by nsheff** [fairtracks/fga-wg#1]
At the meeting on 2024-12-03, we discussed data/file formats. On an Annotation entity, in the data format field, we would like the terms to comply to some vocabulary, and the EDAM 'Sequence annotation track format' could be a possible ontology source for allowed terms for this field. We asked everyone to look at the formats in the current EDAM ontology and see if there is anything missing or needing to be changed.

Those formats are here:

https://bioportal.bioontology.org/ontologies/EDAM?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Fformat_2919

Which looks like this:

![image](https://github.com/user-attachments/assets/22c07adf-06a6-447e-9eb9-029d70b32c95)

**Comment by sunbrn** [comment]
I agree with the flattening of the hierarchy. Formats could be narrow/broadPeaks (any other ones?) and then BEDx+y could be a higher-order "template" for detailed 'bed_types' as @nsheff suggested. 
From EDAM value, I have only worked with 
BED
bedgraph
bigBed
bigWig
MAF
WIG

Are the other ones known? We could make a list of these values and collaboratively add projects/references where they are used so it becomes more apparent which one makes sense to keep (then, we can always leave a "free" value type).

**Comment by sunbrn** [comment]
I would cross check the list from EDAM with the one in Table 2 from https://doi.org/10.1186/1471-2105-12-494

<img width="331" alt="image" src="https://github.com/user-attachments/assets/f72612db-516a-4188-b29d-4a559857587e" />

**Comment by rioualen** [comment]
> I would cross check the list from EDAM with the one in Table 2 

Partially done, I've added it in the new spreadsheet tab that we discussed yesterday

**Comment by rioualen** [comment]
> Here are some suggestions I have:
> 
> * we need short identifiers for formats, maybe corresponding to file extensions, as allowed values. I think 'broadPeak' and 'narrowPeak' are examples
> * there are lots of other possibilities of BED, like: "BED3", "BED3+3", "BED6+2", so really "BEDx+y" should probably be an allowed value. But maybe this is a separate field; maybe there should be 2 fields: for example, 'broadPeak' is a BED6+3 format. So we have one field called "format" and then another called 'bed_type' or 'bed_column_standard' which is the 'BEDx+y' format. So, I think the 'bed6' and 'bed12' are actually a fundamentally different thing, and therefore don't belong here under annotation track formats.
> * I would collapse the hierarchy here; I don't think there needs to be a category for 'ENCODE peak format', these can just live alongside other formats.
> * I don't think 'chrominfo' belongs here.

I agree some reorganisation would be good, however I suggest first focusing on individual terms rather that the overall structure.

It's rather straightforward to add or edit new terms, and allows for their use immediately, while changes in the structure of the ontology are more complex and depend on internal discussions. Besides, future changes in the structure don't affect the usage of the terms (permanent IDs won't change)

**Comment by sveinugu** [comment]
> * we need short identifiers for formats, maybe corresponding to file extensions, as allowed values. I think 'broadPeak' and 'narrowPeak' are examples

I agree. These could be added (if they are not already) as synonyms. The only issue I see here is how to, in a machine-operable way, translate from a term ID to the preferred shorthand label?

> * there are lots of other possibilities of BED, like: "BED3", "BED3+3", "BED6+2", so really "BEDx+y" should probably be an allowed value. But maybe this is a separate field; maybe there should be 2 fields: for example, 'broadPeak' is a BED6+3 format. So we have one field called "format" and then another called 'bed_type' or 'bed_column_standard' which is the 'BEDx+y' format. So, I think the 'bed6' and 'bed12' are actually a fundamentally different thing, and therefore don't belong here under annotation track formats.

Having an explicit `bed_column_standard` field seems too BED-specific for a general schema. But one could possibly add support for properties describing file format-specific variants, which depends on the main file format that is selected, e.g.:

```
file_format: BED
file_format_properties:
  bed_num_core_columns: 6
  bed_num_extra_columns: 3
```

> * I would collapse the hierarchy here; I don't think there needs to be a category for 'ENCODE peak format', these can just live alongside other formats.

Yes, that seems natural. 

> * I don't think 'chrominfo' belongs here.

`chrominfo` is often provided in a BED3 format, but I agree that it is probably not a format per se.
